### PR TITLE
New formatters:  nice_duration() and join_list()

### PR DESCRIPTION
**Pull request by penrods**
Adding two new formatting functions:
* nice_duration(duration, lang="en-us", speech=True)
  Accept seconds or duration and produce a nice sounding duration.
  Example:  nice_duration(61) == "one minute one second"
            nice_duration(61, speech=False) == "1:01"
* join_list(items, connector, sep=None, lang="en-us")
  Example:  join_list(["a", "b", "c"], "and") == "a, b and c"

This includes a translation helper that uses text files in the
mycroft/res/text/LANG/ directory, such as "and.word".  The 
.word files are simple, direct translations -- no randomization.

Also slipped in a .gitignore for the 'dev.env' files that can be used
by VS Code.

## How to test
Run unit tests

### CODE REVIEW

File: mycroft/util/format.py [hunk 1]
```diff
@@ -35,6 +37,34 @@
 import datetime
 import re
 
+
+def _translate(name, lang):
```
Comment (forslund) [inline mycroft/util/format.py]: We should try to use the `mycroft.dialog.get()`, it takes a bit of restructuring to get going, but see the branch [feature/new-formatters](https://github.com/MycroftAI/mycroft-core/tree/feature/new-formatters2)

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
+dev.env
```
[the remark this answers is not in the document]
Comment (penrods) [reply .gitignore]: Whoops, didn't intend to include that in this PR.  That is a file that VS Code uses.

[the remark this answers is not in the document]
Comment (penrods) [reply .gitignore]: Even though this was unintentionally included, I think it is safe to go ahead and pull in.

[the remark this answers is not in the document]
Comment (penrods) [reply .gitignore]: Noted this in the PR

File: mycroft/util/format.py (hunk 1 above)
Comment (penrods) [reply mycroft/util/format.py]: Avoided calling these files .dialog to avoid confusion.  Using .word instead.

**Pull request merged**